tests/TestCase.php

```diff
@@ -35,9 +35,7 @@ abstract class TestCase extends BaseTestCase
     protected function tearDown(): void
     {
         if (null !== $this->previouslyDefinedErrorHandler) {
-            foreach ($this->expectedDeprecations as $expectedDeprecation) {
-                self::assertContains($expectedDeprecation, $this->actualDeprecations);
-            }
+            self::assertSame($this->expectedDeprecations, $this->actualDeprecations);
```

Comment:
this way you depends on order of triggered deprecation. if someone done DX refactoring of method, method works exactly same way, but deprecations will get triggered in other order - this assertion will fail.

Reply:
Is that bad? `SymfonyTestsListener` needs `expectDeprecation` called in order they are triggered.

Reply:
Symfony's implementation aside - why order of deprecations should be interesting? In my opinion you either expect some deprecation or not, it does not matter if the executed code triggers it at the beginning or in the end.

Reply:
I have no problem with not caring about the order (which is how it is after the latest commit), and at the same time, I can imagine someone wanting to check the order in what deprecations were triggered (like maybe from least detailed to most detailed).